build_tools/cmake/static_linker_test.c.in

```diff
@@ -0,0 +1,217 @@
+        iree_runtime_instance_host_allocator(instance), &session);
+  }
+
+  // Load bytecode/c module from the embedded data. Append to the session.
+  iree_vm_module_t* module = NULL;
+
+  if (iree_status_is_ok(status)) {
+    status =
+        create_module(iree_runtime_instance_vm_instance(instance), &module);
+  }
+
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_session_append_module(session, module);
+  }
+
+  // Lookup the entry point function call.
+  const char kMainFunctionName[] = IREE_MODULE_MAIN_FN;
+  iree_runtime_call_t call;
+  memset(&call, 0, sizeof(call));
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_call_initialize_by_name(
+        session, iree_make_cstring_view(kMainFunctionName), &call);
+  }
+
+  // Prepare input.
+  iree_hal_dim_t shape[][IREE_INPUT_DIM_MAX] = IREE_INPUT_SHAPE_ARR;
+  iree_hal_dim_t size[] = IREE_INPUT_SIZE_ARR;
+  iree_hal_dim_t rank[] = IREE_INPUT_DIM_ARR;
+  iree_hal_buffer_view_t* arg_buffer_view = NULL;
+  IREE_INPUT_TYPE* arg_buffers[IREE_INPUT_NUM] = {NULL};
+
+  for (int i = 0; i < IREE_INPUT_NUM; ++i) {
+    if (iree_status_is_ok(status)) {
+      arg_buffers[i] = (IREE_INPUT_TYPE*)aligned_alloc(
+          sizeof(iree_device_size_t), sizeof(IREE_INPUT_TYPE) * size[i]);
+      if (arg_buffers[i] == NULL) {
+        status = iree_make_status(IREE_STATUS_RESOURCE_EXHAUSTED);
+      }
+    }
+    if (iree_status_is_ok(status)) {
+      status = iree_hal_buffer_view_allocate_buffer(
+          iree_hal_device_allocator(device), rank[i], shape[i],
+          IREE_HAL_TYPE, IREE_HAL_ENCODING_TYPE_DENSE_ROW_MAJOR,
+          (iree_hal_buffer_params_t){
+              .type = IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL,
+              .usage = IREE_HAL_BUFFER_USAGE_DEFAULT,
+          },
+          iree_make_const_byte_span((void*)arg_buffers[i],
+                                    sizeof(IREE_INPUT_TYPE) * size[i]),
+          &arg_buffer_view);
+    }
+
+    // Queue buffer views for input.
+    if (iree_status_is_ok(status)) {
+      status =
+          iree_runtime_call_inputs_push_back_buffer_view(&call,
+                                                         arg_buffer_view);
+    }
+    iree_hal_buffer_view_release(arg_buffer_view);
+  }
+
+  // Invoke call.
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_call_invoke(&call, /*flags=*/0);
+  }
+
+  // Cleanup call and buffers.
+  iree_runtime_call_deinitialize(&call);
+  for (int i = 0; i < IREE_INPUT_NUM; ++i) {
+    if (arg_buffers[i] != NULL) {
+      free(arg_buffers[i]);
+    }
+  }
+
+  // Cleanup session and instance.
+  iree_hal_device_release(device);
+  iree_runtime_session_release(session);
+  iree_runtime_instance_release(instance);
+  iree_vm_module_release(module);
+
+  return status;
+}
+
+
+int main() {
+  const char* kName = IREE_EXE_NAME;
+  const iree_status_t result = Run();
+  if (!iree_status_is_ok(result)) {
+    iree_status_fprint(stderr, result);
```

Comment:
Might also want to fprintf stderr the name here ("%s encountered an error:\n")

Reply:
Done.